Alpaca.Markets/PolygonSockClient.cs

```diff
@@ -0,0 +1,274 @@
+                webSocketFactory ?? new WebSocket4NetFactory())
+        {
+        }
+
+        /// <summary>
+        /// Creates new instance of <see cref="SockClient"/> object.
+        /// </summary>
+        /// <param name="keyId">Application key identifier.</param>
+        /// <param name="polygonWebsocketApi">Polygon websocket API endpoint URL.</param>
+        /// <param name="webSocketFactory">Factory class for web socket wrapper creation.</param>
+        public PolygonSockClient(
+            String keyId,
+            Uri polygonWebsocketApi,
+            IWebSocketFactory webSocketFactory)
+        {
+            _keyId = keyId ?? throw new ArgumentException(nameof(keyId));
+
+            polygonWebsocketApi = polygonWebsocketApi ?? new Uri("wss://alpaca.socket.polygon.io/stocks");
+
+            var uriBuilder = new UriBuilder(polygonWebsocketApi)
+            {
+                Scheme = polygonWebsocketApi.Scheme == "http" ? "ws" : "wss"
+            };
+
+            setupWebSocket(uriBuilder, webSocketFactory);
+            _webSocket.Error += handleError;
+            _webSocket.MessageReceived += handleMessageReceived;
+        }
+
+        override private protected JsonAuthRequest getAuthRequest()
+        {
+            return new JsonAuthRequest
+            {
+                Action = JsonAction.PolygonAuthenticate,
+                Params = _keyId
+            };
+        }
+
+        private void handleError(Exception exception)
+        {
+            OnError?.Invoke(exception);
+        }
+
+        private void handleMessageReceived(String message)
+        {
+            try
+            {
+                var rootArray = JArray.Parse(message);
+
+                foreach (JObject root in rootArray)
+                {
+                    var stream = root["ev"].ToString();
+                    switch (stream)
+                    {
+                        case "T":
+                            handleTradeReceived(root.ToObject<JsonStreamTrade>());
+                            break;
+                        case "Q":
+                            handleQuoteReceived(root.ToObject<JsonStreamQuote>());
+                            break;
+                        case "AM":
+                            handleMinuteAggReceived(root.ToObject<JsonStreamAgg>());
+                            break;
+                        case "A":
+                            handleSecondAggReceived(root.ToObject<JsonStreamAgg>());
+                            break;
+                        default:
+                            OnError?.Invoke(new InvalidOperationException(
+                                $"Unexpected message type '{stream}' received."));
+                            break;
+                    }
+                }
+            }
+            catch (Exception exception)
+            {
+                OnError?.Invoke(exception);
+            }
+        }
+
+        private void handleTradeReceived(
```

Comment:
Do we really need all these one-lined methods? Why not just call event handlers in-place using new smart `?.` operator?

Reply:
You're right, it was mostly just because I copied the format from the existing Alpaca class.